release/nightly_tests/dataset/pipelined_inference.py

```diff
@@ -0,0 +1,271 @@
+from collections import OrderedDict
+import argparse
+import os
+import json
+import ray
+import time
+import timeit
+
+import torch.optim as optim
+import numpy as np
+import torch
+import horovod.torch as hvd
+from horovod.ray import RayExecutor
+
+from ray_shuffling_data_loader.data_generation import DATA_SPEC
+from ray_shuffling_data_loader.embedding_model import MyModel, annotation, \
+    huber_loss
+
+# Training settings
+parser = argparse.ArgumentParser(description="Dataset ingestion Example")
+parser.add_argument(
+    "--batch-size",
+    type=int,
+    default=250000,
+    metavar="N",
+    help="input batch size for training (default: 64)")
+parser.add_argument(
+    "--epochs",
+    type=int,
+    default=10,
+    metavar="N",
+    help="number of epochs to train (default: 10)")
+parser.add_argument(
+    "--debug", action="store_true", default=False, help="disables hvd")
```

Comment:
Should we run the tests in this mode to avoid needing GPU?

Reply:
i was thinking we should at least have a small e2e test but larger scale no gpu tests. (like the 500GB window one)

Reply:
Actually one other concern is that using GPUs makes that the bottleneck, so it becomes harder to measure the ingest performance specifically which is more likely to regress.

I'd favor a structure of mostly just no-gpu tests testing pure ingest performance, and then separately a small GPU test just to sanity check GPU functionality.